### ISSUE TITLE
tap2rpm should not use os.system for running other commands

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#4085]
|<img alt="TimAllen's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| TimAllen reported|
|-|-|
|Trac ID|trac#4085|
|Type|enhancement|
|Created|2009-11-12 05:21:28Z|

Currently tap2rpm uses os.system to call various commands including tar and rpmbuild, which has two issues:

 * Little attempt has been made at quoting, which means a single `$` or `"` could make the whole thing blow up.
 * The entire, horribly verbose rpm build process gets dumped to stdout.

tap2rpm should be updated to use Python's `popen2` or `subprocess` modules, or even Twisted's own t.i.utils.getProcessOutputAndValue(). If the return value is zero, the process has probably been successful and the output can be discarded; otherwise the output probably contains useful error messages and should be printed.

<details><summary>Searchable metadata</summary>

```
trac-id__4085 4085
type__enhancement enhancement
reporter__TimAllen TimAllen
priority__normal normal
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1258003288000000 1258003288000000
changetime__1267357879000000 1267357879000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|<img alt="TimAllen's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_1"></a>|TimAllen commented|
|-|-|

This ticket split off from [#3292](https://github.com/twisted/twisted/issues/3292).

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_2"></a>|@therve set status to `closed`|
|-|-|

(In [28572]) Merge modern-tap2rpm-3292 

Authors: TimAllen, jml
Reviewers: glyph, exarkun, therve
Fixes [#3292](https://github.com/twisted/twisted/issues/3292)
Fixes [#4085](https://github.com/twisted/twisted/issues/4085)

Revive the long-dead tap2rpm. This remerge fixes failure with RPM/Tar versions
under Fedora 5.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
